### ISSUE TITLE
NSFW model is loaded over `HTTP` by default, instead of `HTTPS`

### ISSUE THREAD

**Issue by roerohan** [infinitered/nsfwjs#524]
https://github.com/infinitered/nsfwjs/blob/5981af701f650ff57f802e3f2bd980dae9bb63c8/src/index.ts#L30

This does not work on HTTPS websites due to `mixed content` errors.

**Comment by GantMan** [comment]
I'm a bit torn here:

1.  It would be best if you just pull down the model and host it locally.  Rather than depend on the cloudfront version.
2.  What would be a good API?

Thinking out loud:

`load`
```js
await nsfwjs.load('/path/to/model/directory/')
```

`load cloudfront model`
```js
await nsfwjs.infiniteRedHosted()

// AND this for HTTPS?

await nsfwjs.infiniteRedHosted({secure: true})
```

Infinite Red is fine hosting the model for free, but I think sometimes people have no clue they are depending on it.  Maybe a secondary function that is explicit is smart.

**Comment by thebongy** [comment]
I don't think there is any reason to load it over `http` instead of `https` (https is a better default in any case)

On another note, why can't we host the model on github itself, and serve it statically from `https://raw.githubusercontent.com/` ?

**Comment by GantMan** [comment]
Good point.

Would you like to do the PR to make the default https for cloudfront?   Basically a PR adding an 's'?

As for github vs cloudfront, there's definitely a speed difference, but that's a good suggestion.

**Comment by roerohan** [comment]
I've noticed that the CloudFront URL also throws a CORS error when you fix `http` to `https`.

```
Access to fetch at 'https://d1zv2aa70wpiur.cloudfront.net/tfjs_quant_nsfw_mobilenet/model.json' from origin 'https://www.google.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

However, I think the CORS issue won't happen if you host the same thing on `https://raw.githubusercontent.com/`.

**Comment by GantMan** [comment]
My CORS knowledge is a bit shaky.

I tested the HTTPS here:
https://www.test-cors.org/#?client_method=GET&client_credentials=false&server_url=https%3A%2F%2Fd1zv2aa70wpiur.cloudfront.net%2Ftfjs_quant_nsfw_mobilenet%2Fmodel.json&server_enable=true&server_status=200&server_credentials=false&server_tabs=remote

It seems to load with no issue.  Maybe I need more context?

**Comment by roerohan** [comment]
If you use `nsfwjs` in a chrome extension, you'll need to load it over an origin that's not `d1zv2aa70wpiur.cloudfront.net`. For example, if you run the extension on `google.com`, it will try to fetch `model.json` on Google, which is not the same origin as ``, so S3/CloudFront servers will throw a CORS error.

You can simulate the same thing by running a simple `fetch` in your browser's console on `google.com`.

```js
fetch('https://raw.githubusercontent.com/infinitered/nsfwjs/master/src/nsfwjs.ts').then((response) => console.log(response));
```

![image](https://user-images.githubusercontent.com/42958812/123518999-4a19b700-d6c6-11eb-925c-86572208939c.png)

You can fix this by modifying the permissions on the S3 bucket. Once you select your bucket, go to the `Permissions` tab and scroll down to the `Cross-origin resource sharing (CORS)` section. Here you can enter a JSON to define the behaviour you want. Here's an example that allows all origins.

```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]
```

![image](https://user-images.githubusercontent.com/42958812/123519102-e5129100-d6c6-11eb-9ecc-83057df45206.png)
You can also use [this](https://docs.aws.amazon.com/AmazonS3/latest/userguide/cors-troubleshooting.html) for reference.

**Comment by GantMan** [comment]
OK, I want you to file all GitHub tickets from now on :D  This was perfect.

1.  I updated the permissions on the bucket
2.  I tested with the example you gave (from Google) and it now works.

I'm going to add you as a contributor to the project for this insight.   I might still move the code to GitHub, but I want to do that in a separate ticket.

**Comment by GantMan** [comment]
You've been added.   I'll also add the "code" tag if you want to do the PR to switch to https
![image](https://user-images.githubusercontent.com/997157/123519783-f0d46800-d672-11eb-8e1e-fcb2b726b37d.png)

**Comment by roerohan** [comment]
Thanks for adding me as a contributor @GantMan ! 
I've made a PR with the requested change, it will be functional once the CORS issue is fixed.

**Comment by GantMan** [comment]
Please verify the CORS issue is fixed (I verified it did for me), and then you can close this ticket if successful.

**Comment by roerohan** [comment]
Hey @GantMan , I just tried it, I seem to be getting a CORS error still.

![image](https://user-images.githubusercontent.com/42958812/123521865-f6639980-d6d6-11eb-8fa7-ded232ace9d1.png)

**Comment by GantMan** [comment]
I had to do some extra steps: https://stackoverflow.com/questions/12358173/correct-s3-cloudfront-cors-configuration
This means there's a cache that can take a little while.  Try again and let me know.

**Comment by roerohan** [comment]
@GantMan, it is working now. I think it was a caching issue. I'll close this ticket, thanks for your support!
